### PR TITLE
http3: applying config options upstream

### PR DESCRIPTION
**Pull request by alyssawilk**
Risk Level: low (http/3 only)
Testing: new unit tests
Docs Changes: n/a
Release Notes: n/a

### CODE REVIEW

File: source/common/http/http3/conn_pool.h
Comment (RyanTheOptimist) [inline source/common/http/http3/conn_pool.h:22]: Perhaps a comment on this method?

File: source/common/http/http3/conn_pool.h
Comment (RyanTheOptimist) [inline source/common/http/http3/conn_pool.h:51]: Your call, but I'd be inclined to move the body of both the constructor and destructor to the .cc file.

File: test/common/http/http3/BUILD
Comment (RyanTheOptimist) [inline test/common/http/http3/BUILD:15]: nofips? As in the compliance standard?

File: source/common/http/http3/conn_pool.h
Comment (RyanTheOptimist) [inline source/common/http/http3/conn_pool.h:61]: It looks like this method is only called from the constructor. Do we have other places we expect to call it from? If not, perhaps we should just inline it into the constructor?

File: source/common/http/http3/conn_pool.h
Comment (RyanTheOptimist) [inline source/common/http/http3/conn_pool.h:64]: Should this be a const ref, or do we intend callers to modify it?

File: test/common/http/http3/conn_pool_test.cc
Comment (RyanTheOptimist) [inline test/common/http/http3/conn_pool_test.cc:71]: nit: Instead of repeating the numbers, could we do:
options->mutable_max_concurrent_streams()->value()?

**Review by RyanTheOptimist — COMMENTED** [review]
Sweet! Looks great!

File: source/common/http/http3/conn_pool.h
Comment (alyssawilk) [reply source/common/http/http3/conn_pool.h]: done!

File: source/common/http/http3/conn_pool.h
Comment (alyssawilk) [reply source/common/http/http3/conn_pool.h]: done

File: source/common/http/http3/conn_pool.h
Comment (alyssawilk) [reply source/common/http/http3/conn_pool.h]: I think it's slightly cleaner as a utility, but I could make it a static method in the pool to make usage a bit more clear?

File: source/common/http/http3/conn_pool.h
Comment (alyssawilk) [reply source/common/http/http3/conn_pool.h]: this is basically a container of things passed to quic client sessions as they're created.  since many of the items like the crypto config are non-const I think the into needs to be non-const as well.

File: test/common/http/http3/BUILD
Comment (alyssawilk) [reply test/common/http/http3/BUILD:15]: yeah.  they did a fips version of boring a year or two back but we need a more modern version of boringssl for some of the QUIC crypto functions, so our QUIC impl is not compatible with Envoy builds latched to the fips boring.

File: test/common/http/http3/conn_pool_test.cc
Comment (alyssawilk) [reply test/common/http/http3/conn_pool_test.cc]: done

**Review by RyanTheOptimist — DISMISSED** [review]

**Comment by alyssawilk** [comment]
@lizan PTAL

File: source/common/http/http3/conn_pool.h
Comment (lizan) [inline source/common/http/http3/conn_pool.h:21]: `Http3ConnPoolImplTest` is not used in this header?

**Review by lizan — APPROVED** [review]

**Comment by yanavlasov** [comment]
LGTM, modulo small cleanup comment from Lizan.

/wait

**Review by yanavlasov — APPROVED** [review]

**Review by lizan — APPROVED** [review]

**Pull request merged**
